src/java/com/google/cloud/bigquery/dwhassessment/hooks/logger/EventLogger.java

```diff
@@ -168,7 +168,7 @@ private void handleTick() {
   }
 
   private void maybeRolloverWriterForDay() throws IOException {
-    if (writer == null || !logger.getNow().equals(writerDate)) {
+    if (writer == null || recordsWriterFactory.shouldRollover()) {
```

Comment:
Is it possible that the date changes and `writer` is null? If so, the `logFileCount` counter should be set to 1. Currently the date is saved inside `recordsWriterFactory`, while the counter is managed in this class.

I think that coupling together the logic that detects the date change with resetting the counter would make the intent more clear and make the code more readable. The counter could be moved to the `recordsWriterFactory` and the filename could also be constructed there based on the filename pattern.

Reply:
I believe it is possible if retry delay happens at the end of the day. I changed the logic so that log file count resets with every rollover time update. 

I moved all logic for logging state under records write factory, now the log file count increments every time when writer is created. EventLogger only holds UUID, which is generated once per instance.